File: program/find-digital-root-of-a-number/FindDigitalRootOfANumber.cs
using System;
class DigitalRoot {
    static int digroot(int n)
    {
        int root = 0;
        while (n > 0 || root > 9)
        {
            if (n == 0) {
                n = root;
                root = 0;
            }
            root += n % 10;
            n /= 10;
        }
        return root;
    }

    public static void Main()
    {
        Console.Write(digroot(1234));
    }
}

File: program/program/find-digital-root-of-a-number/find_digital_root_of_a_number.pl
sub digroot 
{
    $root = 0;
    while ($n > 0 || $root > 9)
        {
            if ($n == 0) {
                $n = $root;
                $root = 0;
            }
               
            $root += $n % 10;
            $n = $n / 10;
        }
        return $root;
}

$n = 1234;
print digroot(n);
